packages/mobile/src/components/challenge-rewards-drawer/ChallengeRewardsDrawerProvider.tsx

```diff
@@ -124,48 +124,52 @@ export const ChallengeRewardsDrawerProvider = () => {
 
   // Challenge drawer contents
   let contents: Maybe<React.ReactElement>
-  switch (modalType) {
-    case 'referrals':
-    case 'ref-v':
-      contents = (
-        <ReferralRewardContents isVerified={!!config.isVerifiedChallenge} />
-      )
-      break
-    case 'track-upload':
-      contents = config?.buttonInfo && (
-        <Button
-          containerStyle={styles.button}
-          title={config.panelButtonText}
-          renderIcon={config.buttonInfo.renderIcon}
-          iconPosition='right'
-          type={
-            challenge?.state === 'completed' || challenge?.state === 'disbursed'
-              ? ButtonType.COMMON
-              : ButtonType.PRIMARY
-          }
-          onPress={openUploadModal}
-        />
-      )
-      break
-    case 'profile-completion':
-      contents = (
-        <ProfileCompletionChecks
-          isComplete={hasChallengeCompleted}
-          onClose={handleClose}
-        />
-      )
-      break
-    default:
-      contents = config?.buttonInfo && (
-        <Button
-          containerStyle={styles.button}
-          title={config.panelButtonText}
-          renderIcon={config.buttonInfo.renderIcon}
-          iconPosition={config.buttonInfo.iconPosition}
-          type={hasChallengeCompleted ? ButtonType.COMMON : ButtonType.PRIMARY}
-          onPress={handleNavigation}
-        />
-      )
+  if (challenge?.state !== 'completed') {
```

Comment:
this is also truthy if there is no challenge i.e. if challenge is undefined or null. is that what we want?

Reply:
hm no, i should also check that challenge.state is not null/undefined. However, this if statement isn't strictly necessary. thought it would be nice to not go into the switch statement if we know we won't need it (bc only want to show the stuff inside the if statement if the challenge is not completed). thoughts? is this even worth it?

Reply:
if it's not needed then may as well remove it. getting into the switch statement should not be a problem at all

Reply:
hmm but wouldnt the behavior be different? the switch statement would set the `contents` variable if we get in it.